### PR TITLE
Fix #1490

### LINKED ISSUE #1490: Two configurations with different schemas are trying to open the same Realm file.
Hi!

I'm running into a random issue with captioned error when I use two Realms with different configurations (without overlapping modules for each configuration). Any idea for the cause of the error?

This error is not always reproducible and it has appeared under some random occasions on Android 4.3 and 5.1 with 0.82.2.

Thanks for your help!

### PR DESCRIPTION
**Pull request by zaki50**
change the return type of RealmProxyMediator#getModelClasses() to Set in order to guarantee its contents unordered and unique.

TODO

- [x] fix implementation (includes API changes)
- [x] add tests
- [x] update changelog

### CODE REVIEW

File: realm/src/main/java/io/realm/Realm.java [hunk 1]
```diff
@@ -318,10 +318,8 @@ private static Realm createAndValidate(RealmConfiguration configuration, boolean
 
     private static void setupColumnIndices(Realm realm) {
         RealmProxyMediator mediator = realm.configuration.getSchemaMediator();
-        List<Class<? extends RealmObject>> modelClasses = mediator.getModelClasses();
-        int size = modelClasses.size();
-        for (int i = 0; i < size; i++) {
-            Class<? extends RealmObject> modelClass = modelClasses.get(i);
+        Set<Class<? extends RealmObject>> modelClasses = mediator.getModelClasses();
+        for (Class<? extends RealmObject> modelClass : modelClasses) {
```
Comment (emanuelez) [inline realm/src/main/java/io/realm/Realm.java]: Too bad we lose the opportunity to iterate with indices, but it's a one-off operation

Comment (zaki50) [inline realm/src/main/java/io/realm/Realm.java]: We don't have any other choices...

**Comment by zaki50** [comment]
@realm/java please review this

File: changelog.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@
  * Following methods in managed RealmList now throw IllegalStateException instead of ArrayIndexOutOfBoundsException when RealmList.isValid() returns false: set(int,RealmObject), move(int,int), remove(int), get(int)
  * Following methods in managed RealmList now throw IllegalStateException instead of returning 0/null when RealmList.isValid() returns false: clear(), removeAll(Collection), remove(RealmObject), first(), last(), size(), where()
  * RealmPrimaryKeyConstraintException is now thrown instead of RealmException if two objects with same primary key are inserted.
+ * Changed return type of RealmProxyMediator.getModelClasses() to Set<Class<? extends RealmObject>>.
```
Comment (emanuelez) [inline changelog.txt]: In the changelog we only list changes visible to users. So I think you should just state the problem that this PR fixes.

**Comment by emanuelez** [comment]
:+1: apart from the changelog comment

File: realm/src/androidTest/java/io/realm/MediatorTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
+package io.realm;
```
Comment (cmelchior) [inline realm/src/androidTest/java/io/realm/MediatorTest.java]: Missing license header

**Comment by cmelchior** [comment]
Apart from comments this looks good :+1:

File: changelog.txt (hunk 2 above)
Comment (zaki50) [inline changelog.txt]: Oh, I didn't know that RealmProxyMediator is in an internal package.

**Comment by zaki50** [comment]
@emanuelez I updated changelog.txt.

File: realm/src/androidTest/java/io/realm/MediatorTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2014 Realm Inc.
```
Comment (beeender) [inline realm/src/androidTest/java/io/realm/MediatorTest.java]: 2015 Realm Inc.

**Comment by zaki50** [comment]
@emanuelez sure

**Comment by zaki50** [comment]
Oh, 2015...

**Comment by zaki50** [comment]
thanks @beeender

**Comment by dalinaum** [comment]
refs #1490

**Pull request merged**